sdk/core/Azure.Core.Experimental/src/MutableJsonElement.cs

```diff
@@ -405,6 +403,15 @@ private byte[] GetRawBytes()
             return changedElementStream.ToArray();
         }
 
+        internal static Utf8JsonReader GetReaderForElement(JsonElement element)
+        {
+            using MemoryStream stream = new();
+            Utf8JsonWriter writer = new(stream);
+            element.WriteTo(writer);
+            writer.Flush();
+            return new Utf8JsonReader(stream.ToArray());
```

Comment:
you can call GetBuffer().AsSpan().Slice(0, stream.Position) to get the array without a new allocation/copy

Reply:
Excellent - thank you!